pipelines/batch/src/main/java/org/openmrs/analytics/FhirEtl.java

```diff
@@ -201,6 +204,33 @@ static void validateOptions(FhirEtlOptions options) {
 		}
 	}
 	
+	/**
+	 * Driver function for running JDBC direct fetch mode with a HAPI source server. The JDBC fetch mode
+	 * uses Beam JdbcIO to fetch FHIR resources directly from the HAPI server's database and uses the
+	 * ParquetUtil to write resources.
+	 */
+	static void runHapiJdbcFetch(FhirEtlOptions options, DatabaseConfiguration dbConfig, FhirContext fhirContext)
+	        throws PropertyVetoException {
+		FhirSearchUtil fhirSearchUtil = createFhirSearchUtil(options, fhirContext);
+		JdbcConnectionUtil jdbcConnectionUtil = new JdbcConnectionUtil(options.getJdbcDriverClass(),
+		        dbConfig.makeJdbsUrlFromConfig(), dbConfig.getDbUser(), dbConfig.getDbPassword(),
+		        options.getJdbcInitialPoolSize(), options.getJdbcMaxPoolSize());
+		JdbcFetchUtil jdbcFetchUtil = new JdbcFetchUtil(jdbcConnectionUtil);
+		Pipeline pipeline = Pipeline.create(options);
+		
+		String searchUrl = options.getFhirServerUrl() + "/$get-resource-counts";
```

Comment:
Is this a standard FHIR operation or something specific to HAPI? You should be able to achieve what you want with [FHIR standard search params](https://www.hl7.org/fhir/search.html#summary) for summary and count.